src/System.IO.FileSystem/tests/PortedCommon/IOServices.cs

```diff
@@ -111,13 +111,17 @@ public static PathInfo GetPath(string rootPath, int characterCount, int maxCompo
 
         while (path.Length < characterCount)
         {
+            // Add directory seperator after each dir but not at the end of the path
             path.Append(Path.DirectorySeparatorChar);
-            if (path.Length == characterCount)
-                break;
 
             // Continue adding guids until the character count is hit
             string guid = Guid.NewGuid().ToString();
             path.Append(guid.Substring(0, Math.Min(characterCount - path.Length, guid.Length)));
+            if (path.Length + 1 == characterCount)
+            {
+                // If only one character is missing add a k!
+                path.Append('k');
```

Comment:
I though a k would be nice. The character doesn't matter at all because the previous path had tons of guids so there will never be a duplicate because of this character. I though about Random.Next from a to Z boxed to char but it wasn't worth it ;)